### PR TITLE
OCPBUGS-35928: Improve control over PatternFly shared modules used in Console plugins

### PR DESCRIPTION
**Pull request by vojtechszocs**
### 1. Customize `node_modules` directories to search when parsing dynamic modules

New `ConsoleRemotePlugin` option `sharedDynamicModuleSettings.modulePaths` can be used to customize `node_modules` directories to search when parsing dynamic modules. If not specified, the list will contain a single entry (as per existing logic):
```ts
path.resolve(process.cwd(), 'node_modules')
```

Taking [ODF Console Plugin](https://github.com/red-hat-storage/odf-console) as an example, their [webpack config](https://github.com/red-hat-storage/odf-console/blob/master/webpack.config.ts) changes the current work directory to accomodate for different plugin builds:
```ts
const processPath = path.resolve(__dirname, `plugins/${PLUGIN}`);
process.chdir(processPath);
```
However, this breaks the default assumption of vendor packages installed at `{process.cwd}/node_modules` and ultimately results in tons of dynamic module related warnings like:
```
<w> No dynamic module found for Button in @patternfly/react-core
```

To address this issue, plugins can now use the above mentioned option like so:
```ts
new ConsoleRemotePlugin({
  sharedDynamicModuleSettings: { modulePaths: [path.resolve(__dirname, 'node_modules')] }
})
```

### 2. Discard Console provided PatternFly 4.x shared module metadata when using PF 5+

Older plugins using PatternFly **4.x** rely on Console providing the following [shared modules](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/shared-modules-init.ts) at runtime:
- `@patternfly/react-core`
- `@patternfly/react-table`
- `@patternfly/quickstarts`

Newer plugins using PatternFly **5.x** do **NOT** utilize above shared modules. Instead, they share individual PatternFly code bits (components, icons, functions, etc.) via dynamic modules (introduced in #13521).

To improve build performance and allow for smaller production builds, we now discard the Console provided PatternFly 4.x shared module metadata when we detect that the plugin uses PF 5+ (detection is based on `@patternfly/react-core` dependency).

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts [hunk 1]
```diff
@@ -299,21 +329,28 @@ export class ConsoleRemotePlugin implements webpack.WebpackPluginInstance {
...
+      (acc, [pkgName, { indexModule = 'dist/esm/index.js', resolutionField = 'module' }]) => {
+        const basePath = resolvedModulePaths
+          .map((p) => path.resolve(p, pkgName))
+          .find((p) => fs.existsSync(p) && fs.statSync(p).isDirectory());
+
+        if (basePath) {
+          acc[pkgName] = getDynamicModuleMap(basePath, indexModule, resolutionField);
+        }
+
+        return acc;
```
Comment (TheRealJon) [inline frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts:352]: What if we used an immutable accumulator like this:
```suggestion
        return !basePath ? acc : { 
          ...acc, 
          [pkgName]: getDynamicModuleMap(basePath, indexModule, resolutionField)
         };
```

**Review by TheRealJon — APPROVED** [review]
Just one minor nit. Take it or leave it :)

File: frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts (hunk 1 above)
Comment (vojtechszocs) [reply frontend/packages/console-dynamic-plugin-sdk/src/webpack/ConsoleRemotePlugin.ts:352]: @TheRealJon Good idea, I'd personally prefer non-negated variation like
```ts
return basePath
  ? { ...acc, [pkgName]: getDynamicModuleMap(basePath, indexModule, resolutionField) }
  : acc;
```

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**